packages/ui/src/primitives/tailwind/Radio/index.tsx

```diff
@@ -25,82 +25,103 @@ Infinite Reality Engine. All Rights Reserved.
 
 import React from 'react'
 import { twMerge } from 'tailwind-merge'
+import { v4 as uuidv4 } from 'uuid'
 
-export const RadioRoot = ({
-  label,
-  value,
-  onChange,
-  selected,
-  className,
-  labelClassName,
-  disabled,
-  description
-}: {
-  label: string
-  value: string | number
-  onChange: React.ChangeEventHandler<HTMLInputElement>
-  selected: boolean
-  className?: string
-  labelClassName?: string
+import cx from 'clsx'
+
+export interface RadioProps {
   disabled?: boolean
+  defaultChecked?: boolean
+  id: string
+  label: string
+  name?: string
+  onChange?: (event: React.FormEvent<HTMLInputElement>) => void
+  value?: string
   description?: string
-}) => {
-  const twClassname = twMerge('flex flex-col gap-2', className)
-  return (
-    <div className={twClassname}>
-      <label
-        onClick={() => onChange({ target: { value } } as any)}
-        htmlFor={label}
-        className={twMerge('flex cursor-pointer items-center text-sm font-medium text-theme-primary', labelClassName)}
-      >
+}
+
+const RadioRoot = ({ defaultChecked, disabled, id, label, name, onChange, value, description }: RadioProps) => (
+  <div className="flex flex-col">
+    <div className="flex items-start gap-2">
+      <div className="mt-1 grid place-items-center">
         <input
           type="radio"
-          checked={selected}
-          value={value}
-          name={label}
-          onChange={onChange}
+          id={id}
```

Comment:
react provides a new [`useId`](https://react.dev/reference/react/useId) method which can be used for specifying unique ids. lets use that